drivers/lm75/lm75.c

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_lm75
+ *
+ * @{
+ * @file
+ * @brief       Driver for the LM75 temperature sensor.
+ *
+ * A general driver for the lm75 temperature sensor including support for the
+ * lm75a and tmp1075 sensors as well.
+ *
+ * @author      Vitor Batista <vitor.batista@ml-pa.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "lm75.h"
+#include "lm75_regs.h"
+#include "lm75_params.h"
+#include "log.h"
+#include "kernel_defines.h"
+
+#include <stdint.h>
+#include <byteorder.h>
+#include "xtimer.h"
+
+#define I2C_BUS             dev->lm75_params.i2c_bus
+#define I2C_ADDR            dev->lm75_params.i2c_addr
+
+
+int lm75_init(lm75_t *dev, const lm75_params_t *params) {
+
+    dev->lm75_params = *params;
+    uint8_t data;
+    uint8_t config = (params->shutdown_mode) | (params->tm_mode << 1) | (params->polarity << 2) | (params->fault_q << 3);
+
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    /* read the device ID register of the TMP1075 sensor to see if the sensor truly is a TMP1075 */
+    #if MODULE_TMP1075
+        uint16_t deid = 0;
+         if(i2c_read_regs(I2C_BUS, I2C_ADDR, TMP1075_DEVICE_ID_REG, &deid, 2, 0) != 0) {
+            LOG_INFO("Error reading device ID\n");
+         }
+
+        deid = (uint16_t)htons(deid);
+
+        /* checks if the device ID corresponds to the TMP1075 sensor and extends param confi if so*/
+        if (deid == 0x7500) {
+            puts("Device is a TMP1075");
+     		config |= (params->conv_rate << 5) | (params->one_shot_mode << 7);
+   	    	dev->lm75_params.lm75_res.os_res = 625;           /* resolution in 0.0625ºC */
+   	    	dev->lm75_params.lm75_res.os_mult = 10000;        /* Must multiply by 10000 to get temp in ºC */
+   		    dev->lm75_params.lm75_res.os_bits = 4;            /* Only the 12 most significant bits are needed */
+   		    dev->lm75_params.lm75_res.temp_res = 625;         /* resolution in 0.0625ºC */
+   		    dev->lm75_params.lm75_res.temp_mult = 10000;      /* Must multiply by 10000 to get temp in ºC */
+   		    dev->lm75_params.lm75_res.temp_bits = 4;          /* Only the 12 most significant bits are needed */
+        }
+        else {
+            puts("Device ID Register doesnt match");
+            return LM75_ERROR;
```

Comment:
Why not allowing mixing lm75a and tmp1075 devices? We can identify them at run-time after all.